test/framework/src/main/java/org/opensearch/test/ParameterizedDynamicSettingsOpenSearchIntegTestCase.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.test;
+
+import org.opensearch.common.settings.Settings;
+import org.opensearch.common.settings.SettingsModule;
+import org.junit.After;
+import org.junit.Before;
+
+/**
+ * Base class for running the tests with parameterization using dynamic settings: the cluster will be created once before the test suite and the
+ * settings will be applied dynamically, please notice that not all settings could be changed dynamically (consider using {@link ParameterizedStaticSettingsOpenSearchIntegTestCase}
+ * instead).
```

Comment:
should we also add a clarifying java doc here stating that - there will be an update to the settings before and after each parameterized run of the test method. So, every test method inheriting ParameterizedDynamicSettingsOpenSearchIntegTestCase  is run number of times equal to the parameters in one go as opposed to `ParameterizedStaticSettingsOpenSearchIntegTestCase` where all the tests with a single parameter are run at a time and then switched over to the other and so one..

Reply:
Done, thanks @neetikasinghal !

Reply:
Thanks @reta, its much more clearer now